src/main/java/de/tum/in/www1/artemis/service/LearningPathService.java

```diff
@@ -410,4 +411,26 @@ public LearningObject getRecommendation(@NotNull LearningPath learningPath) {
                                 .filter(exercise -> !exerciseRepository.findByIdWithStudentParticipationsElseThrow(exercise.getId()).isCompletedFor(learningPath.getUser()))))
                 .findFirst().orElse(null);
     }
+
+    /**
+     * Gets the health status of learning paths for the given course.
+     *
+     * @param course the course for which the health status should be generated
+     * @return dto containing the health status and additional information (missing learning paths) if needed
+     */
+    public LearningPathHealthDTO getHealthStatusForCourse(Course course) {
+        if (!course.getLearningPathsEnabled()) {
+            return new LearningPathHealthDTO(LearningPathHealthDTO.HealthStatus.DISABLED);
+        }
+
+        var numberOfStudents = userRepository.countUserInGroup(course.getStudentGroupName());
+        var numberOfLearningPaths = learningPathRepository.countByCourseId(course.getId());
```

Comment:
Explicit types would make it a bit clearer, why you compare with equals but then suddenly use a normal - for further calculations.
Why do these methods even return non primitive types?